### ISSUE TITLE
Windows Defender SmartScreen issue

### ISSUE THREAD

**Issue by ww2406** [conda-forge/miniforge#201]
- [X] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Hi all,

Not sure what if anything can be done, but Miniforge's files on Windows keep getting flagged as "dangerous" by SmartScreen. I've tried reporting it as safe twice to Microsoft to seemingly no avail. Is there a digital signature or something that's needed?

**Comment by hmaarrfk** [comment]
seems like we would have to buy a certificate for conda forge.

probably not the end of the world. but it feels like extortion.

https://www.ssl.com/certificates/code-signing/

i wonder if it would even work in this open source setting.

**Comment by traversaro** [comment]
Not sure if @ooeygui have any suggestion w.r.t. this.

**Comment by ooeygui** [comment]
@traversaro I'll take a look.

**Comment by jaimergp** [comment]
I am adding the signing bits for `constructor` now for a separate project. Happy to upstream!

**Comment by jaimergp** [comment]
See https://github.com/conda/constructor/pull/475

**Comment by isuruf** [comment]
@jaimergp, if you are able to do the coding required, I'd be happy to put a vote on conda-forge and buy you a windows certificate under NumFOCUS name. (We already have a macOS certificate under NumFOCUS for conda-forge, but haven't had time to integrate it)

**Comment by jaimergp** [comment]
The PR mentioned above just got merged, so if we are lucky we only need to buy the cert! `main` can now sign both MacOS and Windows. I'd be happy to set up the CI just like we did for `napari`.

**Comment by jaimergp** [comment]
I have added some docs about this in `constructor`: https://conda.github.io/constructor/howto/#signing-and-notarization

Note this [SO answer](https://stackoverflow.com/questions/48946680/how-to-avoid-the-windows-defender-smartscreen-prevented-an-unrecognized-app-fro/66582477#66582477) and this [reddit thread](https://www.reddit.com/r/opensource/comments/nl5u79/is_there_fossfriendly_certificate_authority_to/). 

- Regular code signing certificates aren't the ultimate solution for SmartScreen (it can still get triggered while reputation is built) 
- EV certificates do help better, but they are more expensive and more setup is needed
- We can also [pre-submit things for WDSI analysis](https://www.microsoft.com/en-us/wdsi/filesubmission?persona=SoftwareDeveloper), but they don't have an API 🎉 
- The application process is likely going to be a pain for a distributed organization like conda-forge. NumFOCUS can handle it I guess, but I don't know yet how to handle the physical token parts.

We do want to sign things though (even if it's just for provenance), so certificates we can ask about:

- [Certum](https://shop.certum.eu/open-source-code-signing.html#product.info.customtabs3), cheap but comes with a physical token (see this reddit thread)
- Options mentioned in [this blog post](https://codesigningstore.com/free-code-signing-certificate)

**Comment by rsignell-usgs** [comment]
Has anyone checked the "I am the owner or representative" box?
Perhaps that would help?
I am currently on a visiting scientist assignment and cannot install miniforge3 on the windows machine provided by my organization.


![miniforge3](https://github.com/conda-forge/miniforge/assets/1872600/5ed114de-ff18-4608-955e-c73fbf310400)

**Comment by jaimergp** [comment]
AFAIK, Windows uses a "reputation" score that builds up over time. So there's a chance you can use an older installer that has been "approved" by the reputation system after so many installs. The reputation in the last installer might be insufficient because of its young age. Let me know if that works!

**Comment by rsignell-usgs** [comment]
Thanks @jaimergp !  So I can also ask all my friends to click the "I think this is a safe website" and help the community, right? :)

**Comment by rsignell-usgs** [comment]
A colleague stopped by and I was complaining about this problem and he said "why don't you just use firefox instead of edge?".  
Yep!   I could download just fine with Firefox.  So I'm happy.   :)

**Comment by KhardanOne** [comment]
Hi all,

TL;DR: VirusTotal also complains: https://www.virustotal.com/gui/file/1c107c97a953549be553a09e80b4b32a5132476253606f79282fcd4acbbf36b0

I just run into Mambaforge and also into the SmartScreen issue. For me SmartScreen complains extremely rarely, and in such cases it is usually right. Not always, but mostly. So I checked `Mambaforge-pypy3-Windows-x86_64.exe` and also `Mambaforge-Windows-x86_64.exe` with VirusTotal. Both triggered a warning. See the above report. I suspect it might also be due to the lack of signing. (???) This is the reason I am commenting here instead of opening a separate ticket.

As a comparison, nor Miniconda nor Anaconda installers did not get SmartScreen excited, and Miniconda did not trigger any warnings on VirusTotal.

**Comment by salmiah-ls** [comment]
I download from here https://conda-forge.org/download/ and it works well in my Windows 11.

**Comment by diogo-rossi** [comment]
Hello everyone.
I'm trying to download the installer and the warning is back in the current version.

The images come from MS Edge browser on Windows 10.

Is this ok to download, right?

---

<img width="351" height="69" alt="Image" src="https://github.com/user-attachments/assets/e1b74155-3a54-4011-96a6-b426f5d7c3a8" />

---

<img width="365" height="373" alt="Image" src="https://github.com/user-attachments/assets/1b1069f9-a137-4d39-bf15-e14dd73214ea" />

**Comment by jaimergp** [comment]
Yes, that's the standard message you get when the installer is relatively new and unsigned. Eventually, MS servers get to "know" it as safe and won't flag it, but this one was released only one or two weeks ago so they are still getting to know each other <3

That said, I'm working with NumFOCUS to obtain certificates for signing (we met this week to start the application process for Azure Artifact Signing) so hopefully we can close this issue in this quarter!

**Comment by jaimergp** [comment]
#874 should produce signed EXEs. If someone with Windows can try the CI artifacts there, please let me know 🙏

**Comment by jaimergp** [comment]
Signature can be seen on Windows with [26.3.2-1](https://github.com/conda-forge/miniforge/releases/tag/26.3.2-1) now:

<img width="533" height="500" alt="image" src="https://github.com/user-attachments/assets/7c1036b5-29b6-4d9f-b750-eed8d4c30076" />

<img width="404" height="545" alt="Image" src="https://github.com/user-attachments/assets/bb4c3220-be14-4d37-ab6c-f28d4bdf77fd" />

<img width="404" height="462" alt="Image" src="https://github.com/user-attachments/assets/c67cab34-d5d4-4ae4-a3ec-c207f4ac17fa" />

<img width="403" height="464" alt="Image" src="https://github.com/user-attachments/assets/5f6eda0a-907e-4225-b4e5-57abb9d358ac" />

<img width="405" height="495" alt="Image" src="https://github.com/user-attachments/assets/d2005861-f822-412e-8f71-5d5d88b95f29" />

It still prevented the execution because I assume the certificate is too fresh to have accumulated some reputation. I'll monitor the reports.

**Comment by jaimergp** [comment]
@hmaarrfk can we just change the pre-release to a regular release or does it require a new build?

**Comment by hmaarrfk** [comment]
I think it requires a new build. 

I'm not sure.
